arch/arm/core/aarch32/irq_manage.c

```diff
@@ -253,6 +259,36 @@ int irq_target_state_is_secure(unsigned int irq)
 	return NVIC_GetTargetState(irq) == 0;
 }
 
+/**
+ *
+ * @brief Disable and set all interrupt lines to target Non-Secure state.
+ *
+ * The function is used to set all HW NVIC interrupt lines to target the
+ * Non-Secure state. The function shall only be called fron Secure state.
+ *
+ * Notes:
+ * - All NVIC interrupts are disabled before being routed to Non-Secure.
+ * - Bits corresponding to un-implemented interrupts are RES0, so writes
+ *   will be ignored.
+ *
+ * @return N/A
+*/
+void irq_target_state_set_all_non_secure(void)
+{
+	/* Disable (Clear) all NVIC interrupt lines. */
+	for (uint8_t i = 0; i < sizeof(NVIC->ICER) / sizeof(NVIC->ICER[0]); i++) {
```

Comment:
nit: You use `uint8_t` for your loop index here and below, while you use `int` in `arm_irq_target_state.c`.  IMHO you should use `int` everywhere.

Reply:
sure, I'll do that.

Reply:
done